src/components/LicenseRevokeModal/index.jsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { reduxForm, SubmissionError } from 'redux-form';
+import { Button, Icon, Modal } from '@edx/paragon';
+
+import StatusAlert from '../StatusAlert';
+
+
+class LicenseRevokeModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.errorMessageRef = React.createRef();
+    this.modalRef = React.createRef();
+    this.handleModalSubmit = this.handleModalSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const { current: { firstFocusableElement } } = this.modalRef;
+
+    if (firstFocusableElement) {
+      firstFocusableElement.focus();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      submitFailed,
+      submitSucceeded,
+      onClose,
+      error,
+    } = this.props;
+
+    const errorMessageRef = this.errorMessageRef && this.errorMessageRef.current;
+
+    if (submitSucceeded && submitSucceeded !== prevProps.submitSucceeded) {
+      onClose();
+    }
+
+    if (submitFailed && error !== prevProps.error && errorMessageRef) {
+      // When there is an new error, focus on the error message status alert
+      errorMessageRef.focus();
+    }
+  }
+
+  handleModalSubmit() {
+    const {
+      licenseId,
+      user,
+      sendLicenseRevoke,
+    } = this.props;
+
+    const options = {};
+    options.assignments = [{
+      uuid: user.uuid, email: user.emailAddress, licenseStatus: user.licenseStatus,
+    }];
+
+    return sendLicenseRevoke(licenseId, options)
+      .then((response) => {
+        this.props.onSuccess(response);
+      })
+      .catch((error) => {
+        throw new SubmissionError({
+          _error: [error.message],
+        });
+      });
+    /* eslint-enable no-underscore-dangle */
+  }
+
+  renderBody() {
+    const {
+      user,
+      submitFailed,
+    } = this.props;
+
+    return (
+      <React.Fragment>
+        {submitFailed && this.renderErrorMessage()}
+        <div className="license-details mb-4">
+          <React.Fragment>
+            <p className="message" >Doing so will remove access to the learning catalog and the learner portal
+              for {user.emailAddress}. You will need to add a license to re-enable their learning.
```

Comment:
Let's go with "Revoking a license will remove access to the subscription catalog for {user.emailAddress}. To re-enable access, you can assign this user to another license." for now?